api/resmap/reswrangler.go

```diff
@@ -531,6 +531,20 @@ func (m *resWrangler) appendReplaceOrMerge(res *resource.Resource) error {
 	}
 }
 
+// AnnotateAll implements ResMap
+func (m *resWrangler) AnnotateAll(key string, value string) error {
+	return m.ApplyFilter(
+		kio.FilterFunc(func(nodes []*kyaml.RNode) ([]*kyaml.RNode, error) {
+			for _, rn := range nodes {
+				_, err := rn.Pipe(kyaml.SetAnnotation(key, value))
```

Comment:
it's annoying that the code in filters.annotations.Filter is not the same code (but might do the same thing).  The latter code should probably call kyaml.SetAnnotation to avoid the mystery

Reply:
I didn't know that filters.annotations.Filter existed! I will use that instead.

I think the main difference between kyaml and filter.annotations.Filter is that the latter supports custom fieldspecs, though it can probably be tweaked to both use kyaml and still support custom fieldspecs.